### PR TITLE
Polkadot v0.9.22

### PR DESCRIPTION
**Pull request by zqhxuyuan**
dep:
- [x] orml: https://github.com/open-web3-stack/open-runtime-module-library/pull/745
- [x] xcm-simulator: https://github.com/shaunxw/xcm-simulator/pull/30
- [x] evm-tests: https://github.com/AcalaNetwork/evm-tests/pull/14
- [x] stable-asset: https://github.com/nutsfinance/stable-asset/pull/24

changes:
- [x] token Inspect trait add a mint parameter, default to false.
- [x] migrate jsonrpc to jsonrpsee: https://github.com/paritytech/substrate/pull/8783
- [x] fix inconsist reserve type after: https://github.com/open-web3-stack/open-runtime-module-library/pull/743
- [x] fix evm bench: https://github.com/AcalaNetwork/Acala/pull/2162

### CODE REVIEW

File: modules/evm/rpc/src/lib.rs [hunk 1]
```diff
@@ -450,20 +477,24 @@ where
...
-		let uxt: <B as traits::Block>::Extrinsic = Decode::decode(&mut &*unsigned_extrinsic).map_err(|e| Error {
-			code: ErrorCode::InternalError,
-			message: "Unable to dry run extrinsic.".into(),
-			data: Some(format!("{:?}", e).into()),
+		let uxt: <B as traits::Block>::Extrinsic = Decode::decode(&mut &*unsigned_extrinsic).map_err(|e| {
+			JsonRpseeError::Call(CallError::Custom(ErrorObject::owned(
+				ErrorCode::InternalError.code(),
+				"Unable to dry run extrinsic.",
+				Some(e.to_string()),
+			)))
```
Comment (zjb0807) [inline modules/evm/rpc/src/lib.rs:485]: `format!("execution error: Unable to dry run extrinsic {:?}", e),`

File: modules/evm/rpc/src/lib.rs [hunk 2]
```diff
@@ -450,20 +477,24 @@ where
...
 			.runtime_api()
 			.query_fee_details(&block_id, uxt, unsigned_extrinsic.len() as u32)
-			.map_err(|e| Error {
-				code: ErrorCode::InternalError,
-				message: "Unable to query fee details.".into(),
-				data: Some(format!("{:?}", e).into()),
+			.map_err(|e| {
+				JsonRpseeError::Call(CallError::Custom(ErrorObject::owned(
+					ErrorCode::InternalError.code(),
+					"Unable to query fee details.",
```
Comment (zjb0807) [inline modules/evm/rpc/src/lib.rs:495]: `format!("execution error: Unable to query fee details {:?}", e),`

File: modules/evm/rpc/src/lib.rs [hunk 3]
```diff
@@ -495,10 +526,12 @@ where
...
 
 		let block_limits = if version > 1 {
-			self.client.runtime_api().block_limits(&block_id).map_err(|e| Error {
-				code: ErrorCode::InternalError,
-				message: "Unable to query block limits.".into(),
-				data: Some(format!("{:?}", e).into()),
+			self.client.runtime_api().block_limits(&block_id).map_err(|e| {
+				JsonRpseeError::Call(CallError::Custom(ErrorObject::owned(
+					ErrorCode::InternalError.code(),
+					"Unable to query block limits.",
```
Comment (zjb0807) [inline modules/evm/rpc/src/lib.rs:532]: same here

File: modules/evm/rpc/src/lib.rs [hunk 4]
```diff
@@ -40,52 +44,70 @@ pub use module_evm::{ExitError, ExitReason};
...
+				Err(JsonRpseeError::Call(CallError::Custom(ErrorObject::owned(
+					ErrorCode::ServerError(0).code(),
+					"out of gas".to_string(),
+					None::<()>,
+				))))
+			} else {
+				Err(JsonRpseeError::Call(CallError::Custom(ErrorObject::owned(
+					ErrorCode::InternalError.code(),
+					format!("execution error: {:?}", e),
+					Some("0x".to_string()),
```
[the remark this answers is not in the document]
Comment (zjb0807) [reply modules/evm/rpc/src/lib.rs:94]: I think `internal_err()` can be used to handle the error

**Comment by zqhxuyuan** [comment]
@ermalkaleci could you help fix the bench error?

**Review by xlc — APPROVED** [review]

**Comment by zqhxuyuan** [comment]
@ermalkaleci need fix `evm_gas_limit_test` e2e test as bench changed the weight.

**Comment by ermalkaleci** [comment]
> @ermalkaleci need fix `evm_gas_limit_test` e2e test as bench changed the weight.

we will need to run this again

**Review by xlc — APPROVED** [review]

**Pull request merged**